### ISSUE TITLE
I can insert an image inside a page when adding a page

### ISSUE THREAD

**Issue by djay** [plone/volto#3606]
**Is your feature request related to a problem? Please describe.**
It's confusing that inserting images during add puts those images next to the page rather than inside it. 

**Describe the solution you'd like**
There are a few ways to solve this.
I'd propose an add dialog that lets you set the title and ID and then creates the content before allowing you to edit. Castle CMS did this. 
It's easy to understand and also allows that dialog to do more things like pick a default layout or bulk add content before editing or even upload content. 

**Describe alternatives you've considered**
- make image blocks truely embedded and not create items. https://github.com/plone/volto/issues/1561
- some special temp object (previously done in plone) so no issue with clean up with abandoned. 
- add using temp ID and then rename on first save or delete on cancel. 

**Additional context**

**Comment by tiberiuichim** [comment]
The block transformers were born out of this idea: when creating a new page with an image block, don't immediately upload the image, but send it along with the rest of the page content. In the backend, in the block transformer, detect that a picture data has been sent and "detach" it from the rest of the blocks content, making it a real Image child instance.

**Comment by tiberiuichim** [comment]
The attachments idea has been partially implemented here https://github.com/eea/eea.restapi/tree/master/src/eea/restapi/attachments

**Comment by sneridagh** [comment]
@djay I agree that we should fix this once and for all for Plone. In fact, it's like this because it's what happened in Plone classic since forever.

+1 to @tiberiuichim, not the first time we discuss it, the means to do it is there, we only need a champion.

However, I think it is worth noticing, this issue happens only if the new page is not saved yet. Once you edit, the image gets inside the folderish content type.

**Comment by djay** [comment]
> The block transformers were born out of this idea: when creating a new page with an image block, don't immediately upload the image, but send it along with the rest of the page content. In the backend, in the block transformer, detect that a picture data has been sent and "detach" it from the rest of the blocks content, making it a real Image child instance.

Sounds like a complicated way to solve it.
Sounds more sensible to just only have images attached rather than seperate but that's a seperate debate. 
But there might be content types that have other reasons why they need to insert things inside during adding. 
Question is more what plone really gains by allowing you to edit before the object is created. All it's doing is solving the problem of the object being created empty if you close the battery instead of save. Is that really worth the complexity?

**Comment by djay** [comment]
> However, I think it is worth noticing, this issue happens only if the new page is not saved yet. Once you edit, the image gets inside the folderish content type.

@sneridagh hence why the much simpler way to solve it without fundamentally changing how images work in plone is to do what castleCMS did. Just create the object first before editing. And then you get a lot of bonus benefits (outlined above) in addition that solve other problems that folderish content introduce such how adding a page when viewing a page unexpectedly creates a page inside that page (which is not how plone used to work).

<img width="789" alt="Screen Shot 2022-09-05 at 10 18 19 am" src="https://user-images.githubusercontent.com/41700/188354376-5032dd7e-b55e-4b5a-906d-905a61662ee0.png">
